mautrix_signal/portal.py

```diff
@@ -1321,37 +1322,75 @@ async def _update_participants(self, source: u.User, info: ChatInfo) -> None:
-            await source.sync_contact(address)
-            await puppet.intent_for(self).ensure_joined(self.mxid)
+            try:
+                user = await u.User.get_by_address(address)
+                if user:
+                    members_to_drop.discard(user.mxid)
+                    self.log.debug(
+                        f"_update_participants info.members inviting {user.mxid} to {self.mxid}"
+                    )
+                    if self.config["bridge.noop_invites"] != True:
+                        await self.main_intent.invite_user(self.mxid, user.mxid)
+                puppet = await p.Puppet.get_by_address(address)
+                puppet_intent = puppet.intent_for(self)
+                await source.sync_contact(address)
+                await puppet_intent.ensure_joined(self.mxid)
+                members_to_drop.discard(puppet_intent.mxid)
+            except Exception as ex:
+                self.log.warning(f"failed to join member {address}", ex)
 
         for address in pending_members:
-            user = await u.User.get_by_address(address)
-            if user:
+            try:
+                user = await u.User.get_by_address(address)
+                if user:
+                    members_to_drop.discard(user.mxid)
+                    self.log.debug(
+                        f"_update_participants info.pending_members inviting {user.mxid} to {self.mxid}"
+                    )
+                    if self.config["bridge.noop_invites"] != True:
+                        await self.main_intent.invite_user(self.mxid, user.mxid)
+
+                puppet = await p.Puppet.get_by_address(address)
+                await source.sync_contact(address)
+                pending_mxid = puppet.intent_for(self).mxid
                 self.log.debug(
-                    f"_update_participants info.pending_members inviting {self.mxid} to {user.mxid}"
+                    f"_update_participants info.pending_members inviting {pending_mxid} to {self.mxid}"
                 )
-                if self.config["bridge.noop_invites"] != True:
-                    await self.main_intent.invite_user(self.mxid, user.mxid)
+                members_to_drop.discard(pending_mxid)
+                await self.main_intent.invite_user(self.mxid, pending_mxid)
+            except Exception as ex:
+                self.log.warning(f"failed to invite pending_member {address}", ex)
 
-            puppet = await p.Puppet.get_by_address(address)
-            await source.sync_contact(address)
-            self.log.debug(
-                f"_update_participants info.pending_members inviting {self.mxid} to {puppet.intent_for(self).mxid}"
-            )
-            await self.main_intent.invite_user(self.mxid, puppet.intent_for(self).mxid)
+        if len(members_to_drop) == 0:
+            return
+
+        self.log.warning(f"_update_participants is dropping {members_to_drop} from {self.mxid}")
```

Comment:
```suggestion
        if len(members_to_drop) > 0:
            self.log.warning(f"_update_participants is dropping {members_to_drop} from {self.mxid}")

```

A nitpick, but I feel like the early exit makes the control flow more confusing (and potentially harmful if someone decides to add some more code at the end later and doesn't notice).